pkg/internal/pipe/instrumenter_test.go

```diff
@@ -115,6 +116,41 @@ func TestTracerPipeline(t *testing.T) {
 	matchTraceEvent(t, "GET", event)
 }
 
+func TestTracerReceiverPipeline(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	tc, err := collector.Start(ctx)
+	require.NoError(t, err)
+	consumer := consumer.MockTraceConsumer{Endpoint: tc.ServerEndpoint}
+	require.NoError(t, err)
+	gb := newGraphBuilder(ctx, &beyla.Config{
+		TracesReceiver: beyla.TracesReceiverConfig{
+			Traces: []beyla.Consumer{&consumer},
+		},
+	}, gctx(), make(<-chan []request.Span))
+	// Override eBPF tracer to send some fake data
+	graph.RegisterStart(gb.builder, func(_ traces.ReadDecorator) (node.StartFunc[[]request.Span], error) {
+		return func(out chan<- []request.Span) {
+			out <- newRequest("bar-svc", 1, "GET", "/foo/bar", "1.1.1.1:3456", 404)
+			// closing prematurely the input node would finish the whole graph processing
+			// and OTEL exporters could be closed, so we wait.
+			time.Sleep(testTimeout)
+		}, nil
+	})
+	pipe, err := gb.buildGraph()
+	require.NoError(t, err)
+
+	go pipe.Run(ctx)
+
+	event := testutil.ReadChannel(t, tc.TraceRecords, testTimeout)
+	matchInnerTraceEvent(t, "in queue", event)
+	event = testutil.ReadChannel(t, tc.TraceRecords, testTimeout)
+	matchInnerTraceEvent(t, "processing", event)
+	event = testutil.ReadChannel(t, tc.TraceRecords, testTimeout)
+	matchTraceEvent(t, "GET", event)
```

Comment:
In order to test the attributes passing, would it be possible to verify that some attribute is correct? (for example, the path or the return code).

Reply:
Sorry, I was reading at `matchInnerTraceEvent` but didn't realized that `matchTraceEvent` already does that check.

Then the PR LGTM! (when you make the linter happy)